clang/test/SemaCXX/linkage1.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -pedantic-errors %s
+
```

Comment:
I think this was already pointed out, but some tests w/ `extern "C++"` would be nice, as well as tests for when a namespace is contained in a linkage-spec-decl.